File: TrainWise.MLService/app/models/eda_report.py
from typing import Dict, List, Optional
from pydantic import BaseModel

class OutlierInfo(BaseModel):
    count: int
    percentage: float

class FeatureCorrelation(BaseModel):
    feature: str
    correlation: float

class EdaReport(BaseModel):
    datasetId: str
    targetColumn: Optional[str] = None
    numericColumns: List[str]
    categoricalColumns: List[str]
    duplicateRows: int
    duplicatePercentage: float
    memoryUsageMb: float
    outliersByColumn: Dict[str, OutlierInfo]
    topTargetCorrelations: List[FeatureCorrelation]
    targetDistribution: Optional[Dict[str, int]] = None

File: TrainWise.Web/Models/EdaReport.cs
namespace TrainWise.Web.Models;

public sealed class EdaReport
{
    public string DatasetId { get; set; } = string.Empty;
    public string? TargetColumn { get; set; }
    public List<string> NumericColumns { get; set; } = new();
    public List<string> CategoricalColumns { get; set; } = new();
    public int DuplicateRows { get; set; }
    public double DuplicatePercentage { get; set; }
    public double MemoryUsageMb { get; set; }
    public Dictionary<string, OutlierInfo> OutliersByColumn { get; set; } = new();
    public List<FeatureCorrelation> TopTargetCorrelations { get; set; } = new();
    public Dictionary<string, int>? TargetDistribution { get; set; }
}

public sealed class OutlierInfo
{
    public int Count { get; set; }
    public double Percentage { get; set; }
}

public sealed class FeatureCorrelation
{
    public string Feature { get; set; } = string.Empty;
    public double Correlation { get; set; }
}
